pkg/webhooks/server.go

```diff
@@ -220,8 +208,34 @@ func (ws *WebhookServer) handleAdmissionRequest(request *v1beta1.AdmissionReques
 	}
 	glog.V(4).Infof("Time: webhook GetRoleRef %v", time.Since(startTime))
 
+	// convert RAW to unstructured
+	resource, err := engine.ConvertToUnstructured(request.Object.Raw)
```

Comment:
Pls, use a helper function (with detailed comments) to perform the checks and avoid cluttering the code .